gitlab/client.py

```diff
@@ -1329,4 +1332,28 @@ def execute(
         self, request: Union[str, graphql.Source], *args: Any, **kwargs: Any
     ) -> Any:
         parsed_document = self._gql(request)
-        return self._client.execute(parsed_document, *args, **kwargs)
+        retry = utils.Retry(
+            max_retries=3, obey_rate_limit=True, retry_transient_errors=False
```

Comment:
As a follow-up, we could/should add this to the GraphQL client args. Just trying to keep the diff a bit smaller.